objcgen/processedtypes.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using IKVM.Reflection;
+using Type = IKVM.Reflection.Type;
+
+namespace ObjC {
+
+	// While processing user assemblies, we may come across conditions that will affect
+	// final code generation that we need to pass to the generation pass
+
+	public abstract class ProcessedBase {
+		public bool FallBackToTypeName { get; set; }
+	}
+
+	public class ProcessedMethod : ProcessedBase {
```

Comment:
what if we just named it `Method`, dropping the `Processed` (which sounds like a state)

Might not scale to `Type` :)